sdks/python/apache_beam/io/iobase.py

```diff
@@ -1243,6 +1243,17 @@ def try_claim(self, position):
     """
     raise NotImplementedError
 
+  def is_bounded(self):
+    """Identify whether the output produced by the current restriction is
+    bounded.
+
+    The value is important for the default behavior of truncate when the
+    pipeline starts to drain in streaming. If the current restriction is
+    bounded, it will be processed completely by default. If the restriction is
+    unbounded, it will be truncated into null and finish processing immediately.
+    """
+    raise NotImplementedError
```

Comment:
ThreadsafeRestrictionTracker should call the delegate so that users of it get the same behavior as if they were calling the real restriction tracker

We might want to update the RestrictionTrackerView if we want to expose the boundedness property during execution.